packages/paypal-commerce-integration/src/paypal-commerce-integration-service.ts

```diff
@@ -158,13 +159,22 @@ export default class PayPalCommerceIntegrationService {
         });
     }
 
-    async submitPayment(methodId: string, orderId: string): Promise<void> {
+    async submitPayment(methodId: string, orderId: string, ratePay?: RatePay): Promise<void> {
```

Comment:
I think we should keep this method as abstract as possible, so it might be better to pass an extra date as an argument and move ratepay data structure logic inside ratepay payment strategy